lib/puppet_litmus/rake_helper.rb

```diff
@@ -224,15 +224,26 @@ def install_module(inventory_hash, target_node_name, module_tar, module_reposito
     include ::BoltSpec::Run
 
     target_nodes = find_targets(inventory_hash, target_node_name)
-    bolt_result = upload_file(module_tar, File.basename(module_tar), target_nodes, options: {}, config: nil, inventory: inventory_hash.clone)
+    bolt_result = upload_file(module_tar, "/tmp/#{File.basename(module_tar)}", target_nodes, options: {}, config: nil, inventory: inventory_hash.clone)
     raise_bolt_errors(bolt_result, 'Failed to upload module.')
 
     module_repository_opts = "--module_repository '#{module_repository}'" unless module_repository.nil?
-    install_module_command = "puppet module install #{module_repository_opts} #{File.basename(module_tar)}"
+    install_module_command = "puppet module install #{module_repository_opts} /tmp/#{File.basename(module_tar)}"
     install_module_command += ' --ignore-dependencies --force' if ignore_dependencies.to_s.casecmp('true').zero?
 
     bolt_result = run_command(install_module_command, target_nodes, config: nil, inventory: inventory_hash.clone)
     raise_bolt_errors(bolt_result, "Installation of package #{File.basename(module_tar)} failed.")
+
+    return bolt_result if metadata_module_dependencies.nil?
+
+    metadata_module_dependencies.each do |dependency|
+      module_repository_opts = "--module_repository '#{module_repository}'" unless module_repository.nil?
+      install_module_command = "puppet module install #{module_repository_opts} #{dependency['name']}"
+      install_module_command += " --version '#{dependency['version_requirement']}'" if dependency.key?('version_requirement')
+
+      bolt_result = run_command(install_module_command, target_nodes, config: nil, inventory: inventory_hash.clone)
```

Comment:
the module dependencies are installed along with the module using the original `install_module` command, so what's the reasoning behind adding this?